src/main/java/org/springframework/boot/autoconfigure/data/jdbc/JdbcRepositoriesMyBatisAutoConfiguration.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.boot.autoconfigure.data.jdbc;
+
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.mybatis.spring.boot.autoconfigure.MybatisAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.jdbc.core.DataAccessStrategy;
+import org.springframework.data.jdbc.core.DefaultDataAccessStrategy;
+import org.springframework.data.jdbc.core.SqlGeneratorSource;
+import org.springframework.data.jdbc.mapping.model.JdbcMappingContext;
+import org.springframework.data.jdbc.mybatis.MyBatisDataAccessStrategy;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcOperations;
+
+/**
+ * Autoconfigure Spring Data JDBC
+ *
+ * @author Jens Schauder
+ * @author Greg Turnquist
+ */
+@Configuration
+@ConditionalOnClass(SqlSessionFactory.class)
+@Import(MybatisAutoConfiguration.class)
```

Comment:
This should be unnecessary. Any external autoconfiguration policies are responsible for wiring themselves. Perhaps you actually mean `@AutoConfigureAfter(MybaticAutoConfiguration.class)`, which would be proper.

Reply:
That's something a thought about and couldn't really make up my mind: Should one depend on the starter and let autoconfig make its thing? Or just depend on the autoconfig and call/import it. ... so I guess you are voting for the first.

Reply:
I'm not including the MybatisAutoConfiguration in the tests, since that would require a full configuration of MyBatis (specifying Mappers and stuff).